### PR TITLE
Add search in v-select

### LINKED ISSUE #13712: Complete the Checkbox Tree interface when used in a layout Filter
### Describe the Bug

Complete the Checkbox Tree interface when used in a layout Filter:
Currently, when using the “Equals” operator, a checkbox tree is displayed as an expandable list like this:
image

...but it’s missing the ability to search the tree, which is a critical part of the UI:
image

There is a search at the bottom, but it only searches the raw values, not the labels as configured in the Choices option in the field configuration.

### To Reproduce

New feature request

### Errors Shown

n/a

### What version of Directus are you using?

9.11.1

### What version of Node.js are you using?

Directus Community Cloud

### What database are you using?

SQLite

### What browser are you using?

Chrome

### How are you deploying Directus?

Directus Community Cloud

### PR DESCRIPTION
**Pull request by azrikahar**
## Description

Closes #13712

Todo:

- [x] Only show search input when (nested) items exceeds a threshold

#### Checkbox Tree

https://user-images.githubusercontent.com/42867097/173340547-ea3abddd-82e9-462c-9ec0-b9dbfa66d767.mp4

#### Dropdown with many options (Language)

chrome_VJcXl7xQkW

## Type of Change

- [ ] Bugfix
- [x] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

### CODE REVIEW

**Comment by u12206050** [comment]
Would be great in the filter search if you can toggle the groups to be open if there is only one result.

**Comment by azrikahar** [comment]
> Would be great in the filter search if you can toggle the groups to be open if there is only one result.

Yeah been thinking about it as well, but we'll probably need to explore that after #13752 (specifically v-list-group changes, which is used in select-list-item-group) gets merged to avoid any conflicts there.

**Comment by azrikahar** [comment]
chrome_0qAfm7gjAR

Now it'll auto open when there's one children. However the manual override to close the auto-opened-group(s) will be resolved in #13752 instead.

**Review by rijkvanzanten — APPROVED** [review]

**Comment by bcartier** [comment]
This is a a great start, but there are a couple of major problems:

1. **I can no longer select a parent item in a Checkbox Tree field**. Clicking the label expands/collapses the node, but doesn't allow selection. We'd need allow selection of the node by clicking the label, and expand/collapse of the node by clicking the arrow icon. 

2. **Searching only finds bottom-level nodes in a Checkbox Tree**. Nodes that have children are not found in a search.

**Comment by azrikahar** [comment]
> I can no longer select a parent item in a Checkbox Tree field.

Just to confirm on "no longer", were this possible prior to this PR? At least in the context of this filter value selection of course. Just want to be sure as I believe this PR weren't involved with any relevant components facilitating the group clicks, and it were indeed possible, it could be something affected by other PRs.

> Searching only finds bottom-level nodes in a Checkbox Tree.

Thanks for catching this!

**Comment by bcartier** [comment]
You're right, [azrikahar](https://github.com/azrikahar) - selecting a parent item was not possible before this PR. I just hadn't noticed that issue yet. I updated the comment above to be more accurate. 

I think the selection UI here should work the same way as field/column selection for the table layout with relationship fields - you can select the field name directly, or you can expand the tree to select related fields. For example, I can select **User Created** directly (which then uses the configured Display for the collection), or I can expand **User Created** and select the **Email** field. 

Thanks for looking into it!

**Comment by azrikahar** [comment]
@bcartier thanks for clarifying 👍 totally agree with that. Will follow these up in #14154.

File: app/src/components/v-select/select-list-item-group.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-	<v-list-group :clickable="item.selectable" :value="item.value">
+	<v-list-group :clickable="item.selectable" :open="item.children?.length === 1" :value="item.value">
```
Comment (addisonElliott) [inline app/src/components/v-select/select-list-item-group.vue:2]: @azrikahar Can you explain why you added the `:open` prop?

I use `v-select` on my fork of Directus for a custom use and this line causes issues for me. I have some items with only 1 child and they can't be expanded/collapsed as a result. See the below screenshot.

image

Comment (azrikahar) [reply app/src/components/v-select/select-list-item-group.vue:2]: It was based on [the suggestion above](https://github.com/directus/directus/pull/13867#issuecomment-1153824404), and originally #13752 does make it possible to close it as the logic was within v-list-group! However later on it was moved to v-checkbox-tree, which then subsequently made this change not work as intended anymore as you pointed out 😕

Probably best to remove this line for now, in favor of a potential v-select/v-checkbox-tree refactoring discussed internally in the future.

**Pull request merged**